eng/Versions.props

```diff
@@ -30,11 +30,11 @@
     <!-- Uncomment this line to use the custom version of roslyn as needed. -->
     <!-- <UsingToolMicrosoftNetCompilers Condition="'$(DotNetBuildSourceOnly)' != 'true'">true</UsingToolMicrosoftNetCompilers> -->
     <!-- CoreFX -->
-    <SystemReflectionMetadataVersion>5.0.0</SystemReflectionMetadataVersion>
-    <SystemCollectionsImmutableVersion>6.0.0</SystemCollectionsImmutableVersion>
+    <SystemReflectionMetadataVersion>8.0.0</SystemReflectionMetadataVersion>
```

Comment:
Not something to address or anything. I am not against a bump of these. This does make hard for VS and Roslyn to pick up stuff if we integrate with them in the future.

Reply:
I could have left them alone and backed out the bump Lee did in CLRMD of ...Imutable... It would have made things a lot simpler but it is done now.

Reply:
There's a larger conversation to be had about how long we have to stay stuck on a single build of a dependency.

Since we were forced to make major ClrMD breaking changes, we are now going from ClrMD 3.x -> 4.0  I've used the opportunity to bump the Immutable dependency to 8.0, since there's other breaking changes and a full version bump.

If there's concrete reason we _have_ to stay on 6.0, we can lower ClrMD's Immutable version, but we may be forced to update in the future, and that will require moving ClrMD up to 5.0 instead of 4.0.

I still don't feel like I have a really solid handle on how we are supposed to deal with this as library maintainers.  We can't stay stuck on a single version of dependencies, especially if there's a security incident (unlikely in Immutable, but as a general principle).  I'm not sure when I can move up to current library versions if not when we ship major versions of ClrMD...

Open to suggestions and feedback.  At least until we mark ClrMD as released.  4.0 is currently pre-release and there's still time to change course.